typescript/sdk/src/token/CosmosNativeWarpModule.ts

```diff
@@ -0,0 +1,469 @@
+    const { remoteRouters: expectedRemoteRouters } = expectedConfig;
+
+    const routesToUnenroll = Array.from(
+      difference(
+        new Set(Object.keys(actualRemoteRouters)),
+        new Set(Object.keys(expectedRemoteRouters)),
+      ),
+    );
+
+    if (routesToUnenroll.length === 0) {
+      return updateTransactions;
+    }
+
+    routesToUnenroll.forEach((domainId) => {
+      updateTransactions.push({
+        annotation: `Unenrolling Router ${this.args.addresses.deployedTokenRoute} on ${this.args.chain}`,
+        typeUrl: R.MsgUnrollRemoteRouter.proto.type,
+        value: R.MsgUnrollRemoteRouter.proto.converter.create({
+          owner: actualConfig.owner,
+          token_id: this.args.addresses.deployedTokenRoute,
+          receiver_domain: parseInt(domainId),
+        }),
+      });
+    });
+
+    return updateTransactions;
+  }
+
+  /**
+   * Create a transaction to update the remote routers for the Warp Route contract.
+   *
+   * @param actualConfig - The on-chain router configuration, including the remoteRouters array.
+   * @param expectedConfig - The expected token router configuration.
+   * @returns A array with Cosmos transactions that need to be executed to update the destination gas
+   */
+  async createSetDestinationGasUpdateTxs(
+    actualConfig: HypTokenRouterConfig,
+    expectedConfig: HypTokenRouterConfig,
+  ): Promise<AnnotatedCosmJsNativeTransaction[]> {
+    const updateTransactions: AnnotatedCosmJsNativeTransaction[] = [];
+    if (!expectedConfig.destinationGas) {
+      return [];
+    }
+
+    assert(actualConfig.destinationGas, 'actualDestinationGas is undefined');
+    assert(
+      expectedConfig.destinationGas,
+      'expectedDestinationGas is undefined',
+    );
+    assert(expectedConfig.remoteRouters, 'expectedRemoteRouters is undefined');
+
+    const { destinationGas: actualDestinationGas } = actualConfig;
+    const { destinationGas: expectedDestinationGas } = expectedConfig;
+    const { remoteRouters: expectedRemoteRouters } = expectedConfig;
+
+    // refetch after routes have been previously enrolled without the "actualConfig"
+    // updating
+    const { remote_routers: actualRemoteRouters } =
+      await this.signer.query.warp.RemoteRouters({
+        id: this.args.addresses.deployedTokenRoute,
+      });
+
+    const alreadyEnrolledDomains = actualRemoteRouters.map(
+      (router) => router.receiver_domain,
+    );
+
+    if (!deepEquals(actualDestinationGas, expectedDestinationGas)) {
+      // Convert { 1: 2, 2: 3, ... } to [{ 1: 2 }, { 2: 3 }]
+      const gasRouterConfigs: { domain: string; gas: string }[] = [];
+      objMap(expectedDestinationGas, (domain: string, gas: string) => {
+        gasRouterConfigs.push({
+          domain,
+          gas,
+        });
+      });
+
+      // in cosmos updating the gas config is done by unenrolling the router and then
+      // enrolling it with the updating value again
```

Comment:
what happens when trying to set a value that is already set?

Reply:
it simply won't work (values won't update on chain)